tasks/install-and-configure-packages.yml

```diff
@@ -0,0 +1,27 @@
+- name: Enable standard repositories
+  rhsm_repository:
+    name: "{{ item }}"
+    state: enabled
+  loop: "{{ __ha_cluster_repos }}"
+  when:
+    - ha_cluster_enable_repos
+  ignore_errors: yes
```

Comment:
why do you need to use `ignore_errors: yes` here?

Reply:
It's needed to make the role work in CI systems which don't have access to RHEL subscriptions. This has been added based on a meeting with @pcahyna.

Reply:
If we need some sort of hack, I would prefer to add some sort of extra variable like `ha_cluster_skip_ha_repos` that the CI system could add to tell this task to be skipped.
Or, make it so that the `rhsm_repository` task will see that the CI system has already enabled the necessary repos, so that it is a no-op and will not return an error.
Using `ignore_errors: true` in the actual production code is not good, unless there is just no other solution.

Reply:
There is such a variable: `ha_cluster_enable_repos: no`. I was not aware of the possibility of setting extra vars in CI. If you can do it, then that's a clean solution.

The long term solution, as discussed on the meeting, would be to make this actually work in CI, so the role works without workarounds.

Reply:
> There is such a variable: `ha_cluster_enable_repos: no`. I was not aware of the possibility of setting extra vars in CI. If you can do it, then that's a clean solution.

Well, you can do it, by adding `ha_cluster_enable_repos: no` to `tests/tests_*.yml`

> 
> The long term solution, as discussed on the meeting, would be to make this actually work in CI, so the role works without workarounds.

Reply:
> Well, you can do it, by adding ha_cluster_enable_repos: no to tests/tests_*.yml

If I do it like that, then anyone running the tests would have to enable the repos manually. Is this what we want? My impression was we were trying to work around the issue just in this CI here and let the role enable repos everywhere else. That's why it haven't occur to me to do it like that. But if you think this is the correct way to go, I'm fine with that.

Reply:
> > Well, you can do it, by adding ha_cluster_enable_repos: no to tests/tests_*.yml
> 
> If I do it like that, then anyone running the tests would have to enable the repos manually. Is this what we want?

For now, yes.

> My impression was we were trying to work around the issue just in this CI here and let the role enable repos everywhere else. That's why it haven't occur to me to do it like that. But if you think this is the correct way to go, I'm fine with that.

If we find that this is not suitable, we can change it later.

There's also another problem with `rhsm_repository` - in Ansible 2.10 and later, it looks like it has moved out of core and into community support - https://docs.ansible.com/ansible/latest/collections/community/general/index.html#plugins-in-community-general - which means a dependency on the community.general collection

Furthermore - this so far is the *only* role that has an explicit dependency on an external vendor packaging system - I believe in the other roles, we just mention in the README that there is xxx additional dependency/repo that has to be enabled - so I'm not quite sure how to handle this case.

Reply:
Ok, I've updated the tests so they set `ha_cluster_enable_repos`.

There is an idea to create a separate role for enabling repositories which other roles could use. SAP role also enables repositories and there may be even other roles later which could benefit from that. Personally, I support this idea as that would concentrate all these repositories-related tasks to one place. In the meantime, we can either stick with having `rhsm_repository` in `ha_cluster` role or drop it and note the need for enabling the repo in the README. The readme way is easier for me but I don't feel to be in a position to decide this by myself.

Reply:
Ok.  Let's leave the code as-is with rhsm_repository.  We can revisit this later once we get some feedback from customers, or when we have rhn/rhsm roles.